sprocs/sync_assessments.sql

```diff
@@ -189,6 +195,33 @@ BEGIN
                 student_authz_join = EXCLUDED.student_authz_join,
                 student_authz_leave = EXCLUDED.student_authz_leave,
                 deleted_at = NULL;
+
+            -- Insert all group roles
+            FOR group_role IN SELECT * FROM JSONB_ARRAY_ELEMENTS(valid_assessment.data->'groupRoles') LOOP
+                INSERT INTO group_roles (
+                    role_name,
+                    assessment_id,
+                    minimum,
+                    maximum,
+                    can_assign_roles_at_start,
+                    can_assign_roles_during_assessment
+                ) VALUES (
+                    (group_role->>'role_name'),
+                    new_assessment_id,
+                    -- Insert default values where necessary
+                    CASE WHEN group_role ? 'minimum' THEN (group_role->>'minimum')::integer ELSE 0 END,
+                    CASE WHEN group_role ? 'maximum' THEN (group_role->>'maximum')::integer ELSE (valid_assessment.data->>'group_min_size')::integer END,
+                    CASE WHEN group_role ? 'can_assign_roles_at_start' THEN (group_role->>'can_assign_roles_at_start')::boolean ELSE FALSE END,
+                    CASE WHEN group_role ? 'can_assign_roles_during_assessment' THEN (group_role->>'can_assign_roles_during_assessment')::boolean ELSE FALSE END
```

Comment:
We'd like to have it so instructors don't necessarily have to specify explicitly the minimum and maximum number of group users allowed to be a specific role. In these cases, we want to have the minimum and maximum number default to some reasonable values, e.g. `minimum = 0` and `maximum = the group's minimum size`.

We're not sure if this is the cleanest way to implement this kind of default behavior - what are your thoughts @nwalters512?

Reply:
Ah, I mentioned this above too, but I think we can default to a min of 0 and no max. Is there a certain reason why you were thinking of defaulting the max to the minimum group size?